docs/docs/tutorial/creating-a-source-plugin/part-4/index.mdx

```diff
@@ -0,0 +1,865 @@
+- **Pass options through.** If your source plugin is using a library under the hood (e.g. an SDK from a CMS) that has options on its own, allow people to configure all those options, too. You could collect those under a new key in the `options` object (e.g. `options.sdkOptions`) and then refer people to the SDK's documentation.
+
+### Task: Use `endpoint` option
+
+Instead of hardcoding the URL to the GraphQL API, you'll use an `endpoint` option and thus make the URL configurable.
+
+1. The boilerplate you cloned at the beginning already has an `endpoint` option defined in the site's `gatsby-config`. Go to `site/gatsby-config.ts` to inspect the code:
+
+    ```ts:title=site/gatsby-config.ts
+    import type { GatsbyConfig } from "gatsby"
+    import type { IPluginOptions } from "plugin"
+
+    // highlight-next-line
+    const GRAPHQL_ENDPOINT = `http://localhost:4000/graphql`
+
+    const config: GatsbyConfig = {
+      graphqlTypegen: true,
+      plugins: [
+        {
+          resolve: `plugin`,
+          options: {
+            // highlight-next-line
+            endpoint: GRAPHQL_ENDPOINT,
+          } satisfies IPluginOptions,
+        },
+        `gatsby-plugin-image`,
+        `gatsby-plugin-sharp`,
+        `gatsby-transformer-sharp`,
+      ],
+    }
+
+    export default config
+    ```
+
+    So the plugin already receives the GraphQL endpoint through the `endpoint` option.
+
+1. Update the TypeScript types inside `plugin/src/types.ts` so that both internally and externally the types are correct:
+
+    ```ts:title=plugin/src/types.ts
+    // Rest of types
+
+    interface IPluginOptionsKeys {
+      // highlight-next-line
+      endpoint: string
+    }
+
+    // Rest of types
+    ```
+
+1. Open the file `plugin/src/source-nodes.ts` and validate that you can access the option. You can do this by accessing the second parameter of the `sourceNodes` function and `console.log` it. Use the `IPluginOptionsInternal` type for it:
+
+    ```ts:title=plugin/src/source-nodes.ts
+    // Rest of imports
+    import type {
+      IAuthorInput,
+      IPostInput,
+      NodeBuilderInput,
+      // highlight-next-line
+      IPluginOptionsInternal,
+    } from "./types"
+
+    export const sourceNodes: GatsbyNode[`sourceNodes`] = async (
+      gatsbyApi,
+      // highlight-next-line
+      pluginOptions: IPluginOptionsInternal
+    ) => {
+      const { reporter, cache } = gatsbyApi
+
+      console.log({ pluginOptions })
+
+      // Rest of sourceNodes
+    }
+
+    // Rest of file
+    ```
+
+1. Restart the `develop:site` script and you should see an output in the terminal like this:
+
+    ```shell
+    {
+      pluginOptions: { plugins: [], endpoint: 'http://localhost:4000/graphql' }
+    }
+    ```
+
+    Don't worry about the `plugins` key there. Since Gatsby's plugins can also have sub-plugins, the key `plugins` is added by default to the `pluginOptions` object. The `endpoint` option successfully comes through.
```

Comment:
```suggestion
    Don't worry about the `plugins` key there. Since Gatsby's plugins can also have sub-plugins, the key `plugins` is added by default to the `pluginOptions` object. What you really should care about is that the `endpoint` option successfully comes through.
```